src/handler/http/request/search/job.rs

```diff
@@ -49,3 +70,18 @@ pub async fn query_status(_params: web::Path<String>) -> Result<HttpResponse, Er
 pub async fn query_status(_params: web::Path<String>) -> Result<HttpResponse, Error> {
     Ok(HttpResponse::Forbidden().json("Not Supported"))
 }
+
+#[cfg(feature = "enterprise")]
+async fn cancel_query_inner(trace_ids: &[&str]) -> Result<HttpResponse, Error> {
+    if trace_ids.is_empty() {
+        return Ok(HttpResponse::BadRequest().json("Invalid trace_id"));
+    }
+    let mut res = Vec::with_capacity(trace_ids.len());
+    for trace_id in trace_ids {
+        match crate::service::search::cancel_query(trace_id).await {
+            Ok(status) => res.push(status),
+            Err(e) => return Ok(MetaHttpResponse::bad_request(e)),
```

Comment:
maybe we can skip failed `trace_id` and process the rest and return the result, instead of returning right away?

Reply:
Yep. both are okay.